src/capi.rs

```diff
@@ -738,6 +797,81 @@ pub unsafe extern fn rav1e_twopass_out(ctx: *mut Context) -> *mut Data {
   }))
 }
 
+/// Rate Control Data
+pub enum RcDataKind {
+  /// A Rate Control Summary Packet
+  ///
+  /// It is emitted once, after the encoder is flushed.
+  ///
+  /// It contains a summary of the rate control information for the
+  /// encoding process that just terminated.
+  Summary,
+  /// A Rate Control Frame-specific Packet
+  ///
+  /// It is emitted every time a frame is processed.
+  ///
+  /// The information contained is required to encode its matching
+  /// frame in a second pass encoding.
+  Frame,
+}
+
+/// Return the first pass data
+///
+/// Call it after rav1e_receive_packet().
+///
+/// use rav1e_data_unref() to free the data.
+///
+/// It will return a `RcDataKind::Summary` once the encoder is flushed.
+#[no_mangle]
+pub unsafe extern fn rav1e_rc_receive_pass_data(
+  ctx: *mut Context, data: *mut *mut Data,
+) -> RcDataKind {
+  use crate::api::RcData::*;
+  let (buf, kind) = match (*ctx).ctx.rc_receive_pass_data() {
+    Summary(data) => (data, RcDataKind::Summary),
+    Frame(data) => (data, RcDataKind::Frame),
+  };
+
+  *data = Box::into_raw(Box::new(Data {
+    len: buf.len(),
+    data: Box::into_raw(buf) as *mut u8,
+  }));
+
+  kind
+}
+
+/// Number of pass data packets required to progress the encoding process.
+///
+pub unsafe extern fn rav1e_rc_second_pass_data_required(
+  ctx: *const Context,
+) -> i32 {
```

Comment:
This is not useful. Rav1e does not provide a way to packetize an existing stats file, not does it provide a documentation of the stats file format.

Did we really decide to leave it up to the user to make up their own packetized stats file format so it can read in separate stats packets and their types? That's awful.

Reply:
We did. I/O should not concern the encoder library.

Reply:
It doesn't need to be I/O. It could be a buffer. A user *SHOULD NOT* have to figure out how to packetize and write their data. It should be a buffer that we pass, and the encode consumes, and says how much it consumes.

I have an extremely hard stance on this. It's fricking stupid to force this on a user.

Reply:
I'll document it better, see the test for an example on how to use it.